addons/stock/models/stock_warehouse.py

```diff
@@ -936,11 +936,23 @@ def default_get(self, fields):
     lead_type = fields.Selection(
         [('net', 'Day(s) to get the products'), ('supplier', 'Day(s) to purchase')], 'Lead Type',
         required=True, default='supplier')
+    location_domain = fields.One2many(comodel_name='stock.location', compute='_compute_location_domain')
```

Comment:
allowed_location_ids = fields.One2many('stock.location', compute='_compute_allowed_locations_ids', help='Locations strictly belonging to the selected warehouse)